jscomp/others/js_types.ml

```diff
@@ -63,6 +63,39 @@ let reify_type (type a) (x : 'a) :  (a t * a ) =
   else 
     (Obj.magic Symbol, Obj.magic x) 
   (* TODO: may change according to engines ?*)
+
+type tagged_t = 
+  | JSFalse
+  | JSTrue
+  | JSNull 
+  | JSUndefined     
+  | JSNumber of float 
+  | JSString of string
+  | JSFunction of function_val
+  | JSObject of obj_val
+  | JSSymbol of symbol 
+
+let classify (x : 'a) : tagged_t = 
+  let ty = Js.typeof x in  
+  if ty = "undefined" then 
+    JSUndefined else
```

Comment:
I agree, dangling `else`'s on the end of a line trips me up when reading and refactoring both...